inc/planning.class.php

```diff
@@ -1524,30 +1538,105 @@ static function constructEventsArray($options = []) {
-                           'typeColor'   => (empty($event['event_type_color'])?
-                                             self::getPaletteColor('ev', $event['itemtype']):
-                                             $event['event_type_color']),
-                           'url'         => isset($event['url'])?$event['url']:"",
-                           'ajaxurl'     => isset($event['ajaxurl'])?$event['ajaxurl']:"",
-                           'itemtype'    => $event['itemtype'],
-                           'parentitemtype'    => isset($event['parentitemtype'])?
-                                                   $event['parentitemtype']:"",
-                           'items_id'    => $event['id'],
-                           'priority'    => isset($event['priority'])?$event['priority']:"",
-                           'state'       => isset($event['state'])?$event['state']:""];
+         $new_event = [
+            'title'       => $event['name'],
+            'content'     => $content,
+            'tooltip'     => $tooltip,
+            'start'       => $begin,
+            'end'         => $end,
+            'editable'    => $event['editable'] ?? false,
+            'rendering'   => isset($event['background']) && $event['background']
+                              ? 'background'
+                              : '',
+            'color'       => (empty($event['color'])?
+                              Planning::$palette_bg[$index_color]:
+                              $event['color']),
+            'borderColor' => (empty($event['event_type_color'])?
+                              self::getPaletteColor('ev', $event['itemtype']):
+                              $event['event_type_color']),
+            'textColor'   => Planning::$palette_fg[$index_color],
+            'typeColor'   => (empty($event['event_type_color'])?
+                              self::getPaletteColor('ev', $event['itemtype']):
+                              $event['event_type_color']),
+            'url'         => $event['url'] ?? "",
+            'ajaxurl'     => $event['ajaxurl'] ?? "",
+            'itemtype'    => $event['itemtype'],
+            'parentitemtype' => $event['parentitemtype'] ?? "",
+            'items_id'    => $event['id'],
+            'resourceId'  => $event['resourceId'],
+            'priority'    => $event['priority'] ?? "",
+            'state'       => $event['state'] ?? "",
+         ];
+
+         // override color if view is ressource and category color exists
+         // maybe we need a better way for displaying categories color
+         if ($param['view_name'] == "resourceWeek"
+             && !empty($event['event_cat_color'])) {
+            $new_event['color'] = $event['event_cat_color'];
+         }
+
+         // manage reccurent events
+         if (isset($event['rrule']) && count($event['rrule'])) {
+            $rrule = array_merge($event['rrule'], [
```

Comment:
I had to do this to prevent having an exception:

```php
            $startDt = new \DateTime($new_event['start']);
            $rrule = array_merge($event['rrule'], [
               'dtstart' => $startDt->format('Ymd\THis\Z')
            ]);
```

Exception was: 
```
<br />
<b>Fatal error</b>:  Uncaught RuntimeException: IntlDateFormatter::create() failed. Error Code: 1 &quot;datefmt_create: no such time zone: 'GMT+00:00': U_ILLEGAL_ARGUMENT_ERROR&quot; (this should not happen, please open a bug report!) in /var/www/glpi/vendor/rlanvin/php-rrule/src/RRule.php:2236
Stack trace:
#0 /var/www/glpi/inc/planning.class.php(1588): RRule\RRule-&gt;humanReadable()
#1 /var/www/glpi/ajax/planning.php(43): Planning::constructEventsArray(Array)
#2 {main}
  thrown in <b>/var/www/glpi/vendor/rlanvin/php-rrule/src/RRule.php</b> on line <b>2236</b><br />
```